mods/taskbar-empty-space-clicks.wh.cpp

```diff
@@ -1183,18 +1183,28 @@ static struct
 class COMAPI
 {
 public:
-    COMAPI() : m_isInitialized(false), m_isCOMInitialized(false), m_isUIAInitialized(false), m_isDEInitialized(false),
+    COMAPI() : m_isInitialized(false), m_isCOMInitialized(false), m_needsCOMUninit(false), m_isUIAInitialized(false), m_isDEInitialized(false),
                m_pUIAutomation(nullptr), m_pDeviceEnumerator(nullptr) {}
 
     // init COM for UIAutomation and Volume control
     bool Init()
     {
         if (!m_isCOMInitialized)
         {
-            if (SUCCEEDED(CoInitializeEx(NULL, COINIT_APARTMENTTHREADED)))
+            const auto hr = CoInitializeEx(NULL, COINIT_APARTMENTTHREADED);
+            if (SUCCEEDED(hr))
             {
                 m_isCOMInitialized = true;
-                LOG_INFO(L"COM initilized");
+                if (hr == S_OK) // COM was most likely already initialized in the GUI thread, in case not, we will need to uninitialize it later
+                {
+                    LOG_INFO(L"COM initilized");
+                    m_needsCOMUninit = true;
+                }
+                else
+                {
+                    LOG_INFO(L"COM already initialized");
+                    m_needsCOMUninit = false;
```

Comment:
From [the documentation](https://learn.microsoft.com/en-us/windows/win32/api/combaseapi/nf-combaseapi-coinitializeex):

> To uninitialize the COM library gracefully on a thread, each successful call to [CoInitialize](https://learn.microsoft.com/en-us/windows/desktop/api/objbase/nf-objbase-coinitialize) or CoInitializeEx, including any call that returns S_FALSE, must be balanced by a corresponding call to [CoUninitialize](https://learn.microsoft.com/en-us/windows/desktop/api/combaseapi/nf-combaseapi-couninitialize)

So it was correct before adding `m_needsCOMUninit`.